### ISSUE TITLE
gnome 42 support

### ISSUE THREAD

**Issue by Wings-Fantasy** [mrEDitor/gnome-shell-extension-scroll-panel#21]
Hello developer, Does this extension support gnome 42?

**Comment by mrEDitor** [comment]
Hello! Sorry, I haven't tested it yet, since my Gnome workstations are not up-to-date to Gnome 42.

Thanks to Gnome dev folks for API stability – most probably it will run normally on Gnome 42 if you manually unpack [latest stable release](https://github.com/mrEDitor/gnome-shell-extension-scroll-panel/releases/tag/v4.1.4-gnome41) to ` ~/.local/share/gnome-shell/extensions/io.github.mreditor.gnome-shell-extensions.scroll-panel` and add 42 to `shell-version` list in `metadata.json`.

**Comment by Wings-Fantasy** [comment]
Hello mrEDitor, I restarted gnome-shell after modifying the metadata.json file. Glad to see it working "normally", but in the extension admin panel it says `this._pointerDevice.ungrab is not a function`.
The information displayed after clicking the setting button is:

The settings of extension io.github.mreditor.gnome-shell-extensions.scroll-panel had an error:
```
Error: set_titlebar() is not supported for Adw.Window
```

Stack trace:
```
set_titlebar/<@resource:///org/gnome/Shell/Extensions/js/extensionPrefsDialog.js:51:17
run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:186:20
main@resource:///org/gnome/Shell/Extensions/js/main.js:21:13
run@resource:///org/gnome/gjs/modules/script/package.js:206:19
start@resource:///org/gnome/gjs/modules/script/package.js:190:8
@/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17
```

**Comment by mrEDitor** [comment]
Yes, there were some problems with migrating preferences UI to Gnome Shell 42, now fixed.

You can try out [v4.1.6-gnome42](https://github.com/mrEDitor/gnome-shell-extension-scroll-panel/releases/tag/v4.1.6-gnome42) now.

**Comment by Wings-Fantasy** [comment]
Thanks for the update, it's fully operational

**Comment by ksthiele** [comment]
only thing that is not working properly is the workspace indicator (the dots) are not highlighted while switching workspace

**Comment by mrEDitor** [comment]
@ksthiele thanks for feedback! Are you sure your extension instance is up-to-date? You can see it in extension preferences or `metadata.json` file. Latest version is 4.1.6 for now, worked to me for last time I was on Gnome machine couple of days ago.

I've been expirencing same "not highlighting dots" problem on 4.1.5 (which is not targeted for Gnome Shell 42 — it will not show up preferences for you as well).

If all is up-to-date, please try to temporary disable other extensions, some of them may conflict. If still not helps, I would appreciate if you look for logs (`journalctl /usr/bin/gnome-shell`) if any are produced when you scrolling workspaces.

**Comment by spxak1** [comment]
With version io.github.mreditor.gnome-shell-extensions.scroll-panel v4.1.6+git-6a6efea

On a clean Pop_OS 22.04 it worked fine. Only now it needs a very rapid scroll of the wheel (I use a Logitech MX3 on free wheeling and the equivalent scroll to make the extension work would go over 10-15 pages in 1 second). No updates from the day it worked to when it stopped.

On a upgraded 21.10 to 22.04 (gnome 40 to 42), same version extension, same issue. 

On Fedora 35 to 36, different system/mouse etc, it doesn't work. The extension loads and appears to be fine, but scrolling on the bar won't change workspace anymore. It worked fine on 35.

**Comment by mrEDitor** [comment]
@spxak1, thanks for your feedback! Have you tried to examine and reset extension settings, in each case?

----

The symptoms you described for Pop_OS (I suppose, the thing here is your mouse, not the OS, actually) here can appear because of unsuitable setting `/io/github/mreditor/gnome-shell-extensions/scroll-panel/*-multiplier` values. If reset doesn't make any good, you could also try to set these values manually via
```
dconf write /io/github/mreditor/gnome-shell-extensions/scroll-panel/workspaces-switcher-horizontal-multiplier 1
dconf write /io/github/mreditor/gnome-shell-extensions/scroll-panel/workspaces-switcher-vertical-multiplier 1
```
May be you could get the extension to respond on scrolling more sensitively with fraction values like 0.2 or lesser.
If this helps, I suppose I should change default multiplier settings to lesser values for support of mice with free wheeling.

----

For Fedora, if reset doesn't help, the only think comes to my mind is to fetch and examine debug logs. If you is able to install extension from sources via `DEBUG=1 make install` command, you may try to generate some scrolling events and then fetch debug logs, e.g. via `journalctl /usr/bin/gnome-shell > logfile.txt` command. Debug mode logs really a lot of messages, so when you done you would like to `DEBUG=1 make uninstall` it, most probably.

**Comment by spxak1** [comment]
Thanks for the reply. I have tested on Pop. It works fine one day, then the next it doesn't and I still need to scroll a lot to change desktop (it's unusable really in this mode). 

Changing the scale manually to 1 is the same as resetting (as you said). Changing to 0.2 kills it altogether (it won't change desktop at all as hard as I scroll). 

Enabling the ratchet (rather than free wheeling) makes no difference. Changing the resolution to low on solaar, makes no difference.

I worked for 5 days, now it stopped again. I can't pin point what breaks it. 

I will check Fedora when at work. 

Pop 21.10 and Fedora 35 worked absolutely fine. Same mouse, same version of solaar, same settings. Other extensions of similar functionality also work fine on 22.04 and 36. 

Connecting a bog standard mouse works fine.

Thank you for your time and your work on this extension. I appreciate it.
